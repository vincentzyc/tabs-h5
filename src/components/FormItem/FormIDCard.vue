<template>
  <div class="flex align-middle form-item">
    <div class="form-label">身份证</div>
    <div class="form-input flex-auto flex align-middle">
      <input
        @blur="checkIDCard(model)"
        :maxlength="18"
        placeholder="未满16周岁请勿下单"
        type="text"
        v-model.trim="model"
      />
    </div>
  </div>
</template>

<script lang="ts">
import { checkIDCard } from '@/composition/business/use-verify-data'

export default defineComponent({
  name: "FormIDCard",
  props: {
    modelValue: {
      required: true,
      type: String,
    }
  },
  emits: ['update:modelValue'],
  setup(props, { emit }) {
    const model = computed<string>({
      get() {
        return props.modelValue;
      },
      set(val) {
        emit("update:modelValue", val);
      }
    })
    return {
      model,
      checkIDCard,
    }
  }
})
</script>