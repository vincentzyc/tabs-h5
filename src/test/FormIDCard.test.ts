import { mount } from '@vue/test-utils'
import FormIDCard from '@/components/FormItem/FormIDCard.vue'

test('mount FormIDCard component', async () => {
  expect(FormIDCard).toBeTruthy()

  const wrapper = mount(FormIDCard, {
    props: {
      modelValue: '',
      'onUpdate:modelValue': (e: string) => wrapper.setProps({ modelValue: e })
    }
  })

  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.get('input').setValue('测试身份证号')

  expect(wrapper.props('modelValue')).toBe('测试身份证号')

  const emitEvent = wrapper.emitted('update:modelValue')

  expect(emitEvent).toHaveLength(1)
})