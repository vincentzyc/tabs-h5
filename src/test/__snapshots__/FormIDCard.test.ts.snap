// Vitest Snapshot v1

exports[`mount FormIDCard component 1`] = `
"<div class=\\"flex align-middle form-item\\">
  <div class=\\"form-label\\">身份证号</div>
  <div class=\\"form-input flex-auto flex align-middle\\"><input maxlength=\\"18\\" placeholder=\\"输入办理人身份证（已加密）\\" type=\\"text\\"></div>
</div>"
`;
